src/main/java/net/fabricmc/loader/impl/util/log/ConsoleLogHandler.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2016 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.loader.impl.util.log;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+final class ConsoleLogHandler implements LogHandler {
```

Comment:
Do we need to log this to a file? I imagine there will be some cases where people will provide/go looking for logs that miss imporant details? The vanilla launcher doesnt easily show these errors in the console so it may be impossible to get early logs from it?

Reply:
There is currently no regular logging before the log4j adapter becomes active, but I've added some logic to handle this as best as possible (excl the TODO item in the log4j handler, which is no priority for me).